virttest/qemu_vm.py

```diff
@@ -1567,7 +1567,7 @@ def sort_key(dev):
             devices.insert(serial_devices)
 
             # Create virtio_ports (virtserialport and virtconsole)
-            serial_type = serial_params['serial_type']
+            serial_type = serial_params.get('serial_type', '')
```

Comment:
From my view `serial_type` must be defined when there is `serials` defined, do you have any use case that allow it is empty?

Reply:
if 'serial_type' is mandatory  param , line#2699 should be "if not params.object_params(serial)['serial_type']", base on your origin code, I guess serial_type is not a mandatory param

Reply:
@xutian Em, my fault, it should be mandatory, maybe you could help correct that:), thanks.